test/producer-consumer-spec.js

```diff
@@ -130,13 +166,71 @@ describe('producer/consumer', function () {
       });
     });
 
+    afterEach(() => {
+      cleanupHooks();
+    });
+
     it('should receive message that is only string', async () => {
       const queueName = 'test-only-string-queue';
+      const sentMessage = '85.69.30.121';
+      setupHooks();
 
       await arnavmq.consumer.consume(queueName, (message) => Promise.resolve(`${message}-test`));
       const result = await arnavmq.producer.produce(queueName, '85.69.30.121', { rpc: true });
 
       assert.equal(result, '85.69.30.121-test');
+
+      // Test hooks were called
```

Comment:
This test is a bit bigger, can we have separate tests for each hook (instead of reusing existing test) ?